test/channels.jl

```diff
@@ -323,12 +323,12 @@ end
     ct = current_task()
     testerr = ErrorException("expected")
```

Comment:
Do you think
we should have a custom error type defined in Test stdlib for testing these things,
so  it is harder to get false positives from other errors being thrown?